pkg/minikube/kubeconfig/config.go

```diff
@@ -178,3 +182,58 @@ func decode(data []byte) (*api.Config, error) {
 
 	return config.(*api.Config), nil
 }
+
+// GetKubeConfigStatus verifys the ip stored in kubeconfig.
+// If incorrect, the ip is overwritten.
+func GetKubeConfigStatus(ip string, filename string) (string, error) {
+	if ip == "" {
+		return "", fmt.Errorf("Error, empty ip passed")
+	}
+	ks := ""
+	kip := ""
+	con, err := ReadConfigOrNew(filename)
+	if err != nil {
+		return "", errors.Wrap(err, "Error getting kubeconfig status")
+	}
+	if con.Clusters["minikube"] == nil {
+		return "", errors.Wrap(err, "Kubeconfig does not have a record of a minikube cluster")
+	}
+	kip = con.Clusters["minikube"].Server
+
+	if strings.Contains(kip, ip) {
+		ks = "Correctly configured"
+	} else {
+		ks = "Kubeconfig IP incorrectly configured.  Run minikube update-context to correct."
+	}
+	return ks, nil
+}
+
+// UpdateKubeconfigIP overwrites the IP stored in kubeconfig with the provided IP.
+func UpdateKubeconfigIP(ip string, filename string) (string, error) {
+	if ip == "" {
+		return "", fmt.Errorf("Error, empty ip passed")
+	}
+	con, err := ReadConfigOrNew(filename)
+	if err != nil {
+		return "", errors.Wrap(err, "Error getting kubeconfig status")
+	}
+	if con.Clusters["minikube"] == nil {
+		return "", errors.Wrap(err, "Kubeconfig does not have a record of a minikube cluster")
+	}
+
+	kip := con.Clusters["minikube"].Server
+	ks := ""
+	if strings.Contains(kip, ip) {
```

Comment:
Could you parse this into a URL: https://golang.org/src/net/url/url.go?s=9159:9635#L324

then use the hostname portion for an exact comparison? Bonus points for then parsing that hostname into an IP and using IP.Equal for a better comparison: https://golang.org/pkg/net/#IP.Equal